src/rpm/signature/pgp.rs

```diff
@@ -415,60 +425,32 @@ pub(crate) mod test {
         let data = [1u8; 322];
         let data = &data[..];
 
-        let passwd_fn = String::new;
+        let passwd = Password::empty();
 
         let sig_time = Utc.timestamp_opt(1_600_000_000, 0u32).unwrap();
 
         let mut sig_cfg = SignatureConfig::v4(
             SignatureType::Binary,
             PublicKeyAlgorithm::RSA,
-            HashAlgorithm::SHA2_256,
+            HashAlgorithm::Sha256,
         );
         sig_cfg
             .hashed_subpackets
-            .push(Subpacket::regular(SubpacketData::SignatureCreationTime(
-                sig_time,
-            )));
+            .push(Subpacket::regular(SubpacketData::SignatureCreationTime(sig_time)).unwrap());
         sig_cfg
             .hashed_subpackets
-            .push(Subpacket::regular(SubpacketData::Issuer(
-                signer.secret_key.key_id(),
-            )));
+            .push(Subpacket::regular(SubpacketData::Issuer(signer.secret_key.key_id())).unwrap());
         //::pgp::packet::Subpacket::SignersUserID("rpm"), TODO this would be a nice addition
 
         let signature_packet = sig_cfg
-            .sign(&signer.secret_key, passwd_fn, data)
+            .sign(&signer.secret_key, &passwd, data)
             .expect("Should sign");
 
         signature_packet
             .verify(&verifier.public_key, data)
             .expect("Failed to validate signature");
     }
 
-    #[test]
-    fn verify_subkeys_match() {
-        // verifies that all subkeys are present in both keys under test_assets
-        // which assures all other tests are sane
-        use std::collections::HashSet;
-        let (signer, verifier) = prep();
-        let subkey_set = verifier.public_key.public_subkeys.iter().fold(
-            HashSet::with_capacity(signer.secret_key.public_subkeys.len()),
-            |mut acc, public_subkey| {
-                log::debug!("public subkeys in public key: {:?}", public_subkey.key_id());
-                acc.insert(public_subkey.key_id().as_ref().to_vec());
-                acc
-            },
-        );
-        signer
-            .secret_key
-            .secret_subkeys
-            .iter()
-            .for_each(|public_subkey| {
-                log::debug!("secret subkeys in secret key: {:?}", public_subkey.key_id());
-                assert!(subkey_set.contains(public_subkey.key_id().as_ref()));
-            });
-    }
```

Comment:
@baloo Do you remember why this test was removed

Reply:
pgp changed the layout of its traits and I could not find a way to get that working.